great_expectations/data_context/migrator/cloud_migrator.py

```diff
@@ -143,81 +149,142 @@ def migrate_validation_result(
-            self._break_for_send_configuration_bundle_error(
-                configuration_bundle_response
-            )
-        errors: List[
-            SendValidationResultsErrorDetails
-        ] = self._send_and_print_validation_results(test_migrate)
-        self._print_validation_result_error_summary(errors)
+
         self._print_migration_conclusion_message()
 
-    def _warn_if_test_migrate(self) -> None:
+    def _emit_warnings(
+        self, configuration_bundle: ConfigurationBundle, test_migrate: bool
+    ) -> None:
+        if test_migrate:
+            self._warn_about_test_migrate()
+        if not configuration_bundle.is_usage_stats_enabled():
+            self._warn_about_usage_stats_disabled()
+        if configuration_bundle.datasources:
+            self._warn_about_bundle_contains_datasources()
+
+    def _warn_about_test_migrate(self) -> None:
         pass
 
-    def _warn_if_usage_stats_disabled(self, is_usage_stats_enabled: bool) -> None:
+    def _warn_about_usage_stats_disabled(self) -> None:
         pass
 
-    def _build_configuration_bundle(self) -> ConfigurationBundle:
+    def _warn_about_bundle_contains_datasources(self) -> None:
         pass
 
-    def _print_configuration_bundle(
+    def _print_configuration_bundle_summary(
         self, configuration_bundle: ConfigurationBundle
     ) -> None:
         pass
 
-    def _send_configuration_bundle(
+    def _serialize_configuration_bundle(
+        self, configuration_bundle: ConfigurationBundle
+    ) -> dict:
+        serializer = ConfigurationBundleJsonSerializer(
+            schema=ConfigurationBundleSchema()
+        )
+        serialized_bundle = serializer.serialize(configuration_bundle)
+        return serialized_bundle
+
+    def _send_configuration_bundle(self, serialized_bundle: dict) -> None:
+        print("Sending context configuration (step 2/4)")
+        response = self._post_to_cloud_backend(
+            resource_name="migration",
+            resource_type="migration",
+            attributes_key="bundle",
+            attributes_value=serialized_bundle,
+        )
+        # TODO: Handle success/failure cases
+
+    def _prepare_validation_results(self, serialized_bundle: dict) -> List[dict]:
+        print("Preparing validation results (step 3/4")
+        return serialized_bundle.pop("validation_results")
+
+    def _send_validation_results(
+        self, serialized_validation_results: List[dict]
+    ) -> None:
+        # 20220928 - Chetan - We want to use the static lookup tables in GeCloudStoreBackend
+        # to ensure the appropriate URL and payload shape. This logic should be moved to
+        # a more central location.
+        resource_type = GeCloudRESTResource.EXPECTATION_VALIDATION_RESULT
+        resource_name = GeCloudStoreBackend.RESOURCE_PLURALITY_LOOKUP_DICT[
+            resource_type
+        ]
+        attributes_key = GeCloudStoreBackend.PAYLOAD_ATTRIBUTES_KEYS[resource_type]
```

Comment:
feels icky

Reply:
What if there was a `post_to_cloud_backend()` public method on the GeCloudStoreBackend that we used here? Could also be used internally in the GeCloudStoreBackend. We could create a GeCloudStoreBackend in the CloudMigrator init flow and use that internally here to send the bundle/validations.

Reply:
this could be useful! I'd want to use the same session object for all requests though - I'd have to think about how to go about this

Reply:
think I'm going to table this for the time being but come back to it if that's alright